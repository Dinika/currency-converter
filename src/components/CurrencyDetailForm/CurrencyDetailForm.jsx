import React from 'react';
import { Form } from 'semantic-ui-react';

const CurrencyDetailForm = (props) => {
    return (
        <Form>
            <Form.Group widths='equal'>
                <Form.Field placeholder='0.00' control='input' />
                <Form.Field control='select'>
                    <option value='EUR'>EUR</option>
                    <option value='USD'>USD</option>
                    <option value='JPY'>JPY</option>
                </Form.Field>
            </Form.Group>
        </Form>
    )
}

export default CurrencyDetailForm;
