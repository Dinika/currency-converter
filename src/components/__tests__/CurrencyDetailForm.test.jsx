import React from 'react';
import { shallow } from 'enzyme';

import CurrencyDetailForm from '../CurrencyDetailForm/CurrencyDetailForm';

describe('CurrencyDetailForm', () => {
  it('renders with all props set', () => {
    const currencyOptions = ['Option1', 'Option2'];

        
const currencyAmount = 10;        
    const wrapper = shallow(
        <CurrencyDetailForm
            currencyOptions={currencyOptions}
            currencyAmount={currencyAmount}
            onCurrencyAmountChange={jest.fn()}
            onCurrencyTypeChange={jest.fn()} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});