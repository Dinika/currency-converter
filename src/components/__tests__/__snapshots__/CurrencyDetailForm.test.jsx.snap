// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyDetailForm renders with all props set 1`] = `
<Form
  as="form"
>
  <FormGroup
    widths="equal"
  >
    <FormField
      control="input"
      onChange={[MockFunction]}
      placeholder={10}
      type="number"
      value={10}
    />
    <FormField
      control="select"
      onChange={[MockFunction]}
    >
      <option
        key="Option1"
        value="Option1"
      >
        Option1
      </option>
      <option
        key="Option2"
        value="Option2"
      >
        Option2
      </option>
    </FormField>
  </FormGroup>
</Form>
`;
